### PR TITLE
Lua Files for Rostrum Pump NM

### PR DESCRIPTION
**Pull request by nesstea**
Some reason they didn't commit or weren't merged.

### CODE REVIEW

File: scripts/zones/Attohwa_Chasm/mobs/Feeler_Antlion.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+require("scripts/globals/status");
+require("scripts/globals/magic");
+
+-----------------------------------
+-- onMobSpawn Action
+-----------------------------------
+
+function onMobSpawn(mob)
+	mob:addMod(MOD_REGAIN, 40);	-- Don't know exact value
+	mob:addMod(MOD_REGEN, 30);
```
Comment (xdemolish) [inline scripts/zones/Attohwa_Chasm/mobs/Feeler_Antlion.lua]: addMod on initialise, not spawn

Comment (nesstea) [inline scripts/zones/Attohwa_Chasm/mobs/Feeler_Antlion.lua]: They don't need to be applied until the mobs are spawned.

File: scripts/zones/Attohwa_Chasm/mobs/Executioner_Antlion.lua [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+require("scripts/globals/status");
+require("scripts/globals/magic");
+
+
+-----------------------------------
+-- onMobInitialize Action
+-----------------------------------
+
+function onMobInitialize(mob)
+	mob:setTP(100);
```
Comment (teschnei) [inline scripts/zones/Attohwa_Chasm/mobs/Executioner_Antlion.lua]: this doesn't do anything

File: scripts/zones/Attohwa_Chasm/mobs/Feeler_Antlion.lua (hunk 1 above)
Comment (teschnei) [inline scripts/zones/Attohwa_Chasm/mobs/Feeler_Antlion.lua]: put it in initialize

File: scripts/zones/Attohwa_Chasm/mobs/Alastor_Antlion.lua [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+-----------------------------------
+-- onAdditionalEffect
+-----------------------------------
+
+function onAdditionalEffect(mob, player)
+    local chance = 25;
+	local resist = player:getMod(MOD_PETRIFYRES);
+    local duration = 30; -- Don't know the actual duration.
+	if (resist>0) then
```
Comment (teschnei) [inline scripts/zones/Attohwa_Chasm/mobs/Alastor_Antlion.lua]: use the function in magic.lua to figure out resistance checks

Comment (nesstea) [inline scripts/zones/Attohwa_Chasm/mobs/Alastor_Antlion.lua]: But I am not casting a spell. It's the onAdditionEffect.

Comment (TeoTwawki) [inline scripts/zones/Attohwa_Chasm/mobs/Alastor_Antlion.lua]: @nesstea look at existing mob add effect scripts...They use the function he pointed you at.

Comment (teschnei) [inline scripts/zones/Attohwa_Chasm/mobs/Alastor_Antlion.lua]: a status effect, regardless of where it comes from, is still subject to resistances in the same was as the ones cast via spell

File: scripts/zones/Attohwa_Chasm/mobs/Alastor_Antlion.lua [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+require("scripts/globals/magic");
+
+
+-----------------------------------
+-- onMobInitialize Action
+-----------------------------------
+
+function onMobInitialize(mob)
+	mob:setMobMod(MOBMOD_ADD_EFFECT,mob:getShortID());
+	mob:setTP(100);
```
Comment (TeoTwawki) [inline scripts/zones/Attohwa_Chasm/mobs/Alastor_Antlion.lua]: setTP() won't work in onMobInitialize, you'll have to put it in onMobSpawn if you want it spawned with TP.

**Comment by maxtherabbit** [comment]
copying my comment from the additional effect weapons pull request:

I will be shortly modifying applyResistanceAddEffect() to apply specific status resistance bonuses. With this in mind, the correct format for the call is: applyResistanceAddEffect(player,target,ELE_WHATEVER,EFFECT_WHATEVER)

The 4th param is currently an unused param for a bonus, so I ask that any new additional status effect scripts added to the trunk replace the '0' with the appropriate EFFECT_X as of now :D

**Comment by nesstea** [comment]
I'll change it, but barpetra doesnt matter in this state at all.

**Pull request merged**